### ISSUE TITLE
[Feature Request] Publish the android app

### ISSUE THREAD

**Issue by MohamedBassem** [MohamedBassem/hoarder-app#6]
The mobile app is already cross-platform. We just need to test and publish it to the play store.

## EDIT:

Ok, so google just accepting the app for closed testing!

To join the closed testing group, you'll need to join this google group: https://groups.google.com/g/hoarder-android-testers

Once you've joined (I guess you might need to give it a minute or two), you can use the following link to install the app from the app store:

- If you're on the browser, you can join the beta testing program from that link: https://play.google.com/apps/testing/app.hoarder.hoardermobile
- and from android, using that link: https://play.google.com/store/apps/details?id=app.hoarder.hoardermobile

We will need at least 20 testers, for 2 weeks for the app to get listed on the play store. Please give it a try and let me know how it goes!

**Comment by runinit** [comment]
I can test this if needed

**Comment by MohamedBassem** [comment]
As of (https://github.com/MohamedBassem/hoarder-app/commit/26b53e2ccc00befd182c4af05ab52fc439be7535), the android app is ready for submission to the play store. Going through the submission process right now :)

**Comment by MohamedBassem** [comment]
Ok, the app right now is under google's review. Will share the testing link once it's available. We'll need to get some testers before it's widely distributed in the play store.

**Comment by DavidN0809** [comment]
I am open to testing as well

**Comment by mp-strachan** [comment]
Also happy to opt in for testing

**Comment by iconoclasthero** [comment]
If the .apk was linked in this thread, i'd put it on _a_ phone I have...

**Comment by MohamedBassem** [comment]
Here's the apk: https://drive.google.com/file/d/1VEeN6KDCHe0jy8PmX0I3sZLtY5DZ2Wd3/view?usp=sharing (first time doing this, so hope I've done it right).

Will share the testing invitation link once google allows me to do so.

**Comment by iconoclasthero** [comment]
```
Network request failed
Server Address
192.168.1.2:3004
Email
iconoclasthero@anemailsite.com
Password
********
```

**Comment by MohamedBassem** [comment]
You probably should add the protocol as well, so `http://192.168.1.2:3004` (or https if you're using https)

**Comment by iconoclasthero** [comment]
Settings > Log > Logs me out instead of showing log.

**Comment by MohamedBassem** [comment]
@iconoclasthero thanks a lot for the feedback :) You're the first android tester, so expect some bugs :) Please keep the bug report coming.

1. I'll look into the weird behavior of the server address. I think I know what's happening.
2. I'll look into the auto captilize thing as well.
3. `Settings > Log`. There's no `Log` it's actually a `Log out` button :D 

<img width="425" alt="Screenshot 2024-03-30 at 3 27 54 PM" src="https://github.com/MohamedBassem/hoarder-app/assets/2418637/c2d02be9-a9ef-4b10-9ab5-7b97ef7b1091">

**Comment by iconoclasthero** [comment]
I call bullshit!  I found the `Log` button!
![47f92e4e-2fe7-440f-998c-afd43685103f](https://github.com/MohamedBassem/hoarder-app/assets/535962/bea57864-1bdb-4d01-b36e-36c3ede25987)

**Comment by MohamedBassem** [comment]
wtf, how did this happen. Mobile development is so weird 😂

**Comment by iconoclasthero** [comment]
On the plus side, here's how I shared that image... I took a screenshot inside the android app, went to share the image from the screenshot dialog, and the button f/hoarder came up...which i hit.  I went to the pc, had to hit refresh, and it was only a white background with no information, but it was there.

**Comment by MohamedBassem** [comment]
You having to hit refresh is tracked in https://github.com/MohamedBassem/hoarder-app/issues/29.

It being white, I think have something to do with how I "stretch" the content in the container and the screenshot being mostly white. I think if you click the maximize button, you'll see the full screenshot.

Typically, there would be tags, but judging from your other bookmarks, it seems that you don't have automatic tagging enabled. That's why the card feels so "empty" 😅

**Comment by iconoclasthero** [comment]
Well, this is bluesky, but i'd want "something" to look at the image and find the "most interesting" part of it and feature that.  How you do that, not sure.  I think it's possible to set global options for how images are displayed, i.e., are they thumbed and centered, stretched, ...  and then as this develops, that should change to being able to individually set how those images are shown.

**Comment by JFPCreations** [comment]
Thanks for the app. So far so good. No problem found.

**Comment by Reledia** [comment]
I'm trying the app on android and I can find no issues with it. I would only suggest to implement a re-fetch button in the note menu for when inference fails. (And possibly even a tag/error that notify of this)

**Comment by iconoclasthero** [comment]
> Typically, there would be tags, but judging from your other bookmarks, it seems that you don't have automatic tagging enabled. That's why the card feels so "empty" 😅 I'm open to suggestions on how to make it more better looking.

I wanted to ask about that...is there a discord or anything?

![Screenshot from 2024-03-30 13-02-34](https://github.com/MohamedBassem/hoarder-app/assets/535962/302beec1-7b60-4862-a3df-8cc50f4a9eb2)

**Comment by MohamedBassem** [comment]
I've created a discord channel: https://discord.gg/NrgeYywsFh

Let's chat there.

**Comment by lucius346346** [comment]
> ```
> Network request failed
> Server Address
> 192.168.1.2:3004
> Email
> iconoclasthero@anemailsite.com
> Password
> ********
> ```


I'm having the same error but it persist. Also same error on Firefox addon (NetworkError when attempting to fetch resource). Is it some misconfiguration on my part? I've tried it with  or without http://.

**Comment by MohamedBassem** [comment]
@lucius346346 check https://github.com/MohamedBassem/hoarder-app/issues/23 for the firefox problems.

As for the app, what errors are you getting?

**Comment by lucius346346** [comment]
> @lucius346346 check #23 for the firefox problems.
> 
> As for the app, what errors are you getting?

It's "Network request failed" no matter what I type, including blank. I'm sure I'm typing the right credentials - I can log with them via browser.

**Comment by MikeKMiller** [comment]
Same here with the android app.

**Comment by stanrc85** [comment]
Same here, tried both with ip and hostname

**Comment by MohamedBassem** [comment]
@MikeKMiller @stanrc85 any chance any of you is using a self-signed (or invalid) https cert?
Btw, if any of you would like to hope onto Hoarder's discord so that we can debug it together, I'd be thankful.

**Comment by stanrc85** [comment]
No https for me, I posted in the discord!

**Comment by MikeKMiller** [comment]
I have no https either

**Comment by lucius346346** [comment]
Also, no.

**Comment by MohamedBassem** [comment]
I managed to repro locally, and the good news is that I think I found the problem. Apparently android release builds by default refuse to connect to HTTP endpoinds (requiring https basically). This behavior is different in release builds vs development builds and hence why I didn't catch it during development.

I'll try some of the mitigations I googled and report back.

**Comment by MohamedBassem** [comment]
I checked immich for example and it seems to be setting this `usesCleartextTraffic=true`. That I hope will be the fix. Trying it now.

https://github.com/immich-app/immich/blob/861b72ef0493bb6272c1a1b16e5cdee6792fb4b6/mobile/android/app/src/main/AndroidManifest.xml#L3

**Comment by MikeKMiller** [comment]
I was just going to suggest that.

**Comment by MohamedBassem** [comment]
Good news, it worked! At least on my test phone!

I've updated the APK link with the new version:

https://drive.google.com/file/d/1VEeN6KDCHe0jy8PmX0I3sZLtY5DZ2Wd3/view?usp=sharing

Please give it a try and let me know :)

**Comment by MikeKMiller** [comment]
That worked, I can login now.

**Comment by stanrc85** [comment]
Success!

**Comment by MohamedBassem** [comment]
Ok, so google just accepting the app for closed testing!

To join the closed testing group, you'll need to join this google group: https://groups.google.com/g/hoarder-android-testers

Once you've joined (I guess you might need to give it a minute or two), you can use the following link to install the app from the app store:

- If you're on the browser, you can join the beta testing program from that link: https://play.google.com/apps/testing/app.hoarder.hoardermobile
- and from android, using that link: https://play.google.com/store/apps/details?id=app.hoarder.hoardermobile

We will need at least 20 testers, for 2 weeks for the app to get listed on the play store. Please give it a try and let me know how it goes!

**Comment by woaloo** [comment]
Hi,
I've just install this app with my self-hosted Hoarder instance. Woaw, it's working great !
Thanks a lot for your job

**Comment by Batman313v** [comment]
I didn't see any conversations in the google group so I'll just mention it here, Please add dark mode. I installed and opened the app (it's midnight local time) and burned my bookmarks to my retinas 😂 Other than that, it'd be nice to have one of the following options:   

Multiple "Share" options; a quick one (how it is now) and one that pulls up a menu to add tags.  

A setting to enable a "complex" share menu with that feature 😁

(Please note: not complaining just giving feedback as a early tester. Love the project, great app 👍)

**Comment by MohamedBassem** [comment]
@Batman313v I've just landed dark mode in https://github.com/MohamedBassem/hoarder-app/commit/c46482cdaaf883971736488750513663dd023076 ! I'll cut a new mobile release now and it should be available on your phone when google approves the new release!

As for the complex share menu, this is planned.

**Comment by MohamedBassem** [comment]
Dark mode show now be available in version `1.4.0`.

As for publishing the app to the play store. According to google, we're at day 9 of 20. 11 more days to go before we're able to publish the app directly to the play store!

**Comment by mycodex-dev** [comment]
Just installed the Android app and entering the server address on the login screen has a behavior where Gboard hides after each key press 

h, keyboard hides, t, keyboard hides, t, keyboard hides, p, keyboard hides...

I had to type the server address in another app and copy/paste into the server address field in your app because it would have taken too long.

**Comment by MohamedBassem** [comment]
@mycodex-dev I've just now released v1.5 for the mobile app (might take an hour until google propagates it), that fixes this particular issue. Sorry for the hassle.

**Comment by mycodex-dev** [comment]
> @mycodex-dev I've just now released v1.5 for the mobile app (might take an hour until google propagates it), that fixes this particular issue. Sorry for the hassle.
> 
> EDIT: Commit for reference: [4f9b6c8](https://github.com/MohamedBassem/hoarder-app/commit/4f9b6c89dedbe48ced4ea3448a71a75714045453)

Awesome! Love the mobile app so far. This completely removed my need for Pocket and even Google Keep with the notes functionality with markdown support *chef's kiss*. 

Keep up the great work!

**Comment by MohamedBassem** [comment]
Hey folks, thanks a lot for your support! Hoarder's android app is [now live on the play store!](https://play.google.com/store/apps/details?id=app.hoarder.hoardermobile&pcampaignid=web_share). If you're interested in more stability, you can now switch to the prod app, if you're interested for the latest updates, you can stay on the testing list. And while, you're at it, maybe maybe leave a review on the play store ;)
Thanks!
